Makefile

```diff
@@ -80,7 +80,18 @@ ifneq ($(FULL_CPU),)
     endif
 endif
 
+# On our default platform, and CPU, we need a minimal CPU variation to fit,
+# so we default the variant to minimal on the default platform; on any other
+# platform we default to the standard CPU.  We want this to end up in
+# the prompt, so we do not set it as the DEFAULT_CPU_VARIANT, but apply
+# it later.
+#
 CPU ?= $(DEFAULT_CPU)
+ifeq ($(PLATFORM),$(DEFAULT_PLATFORM))
```

Comment:
I don't think we need to do this... Let me look closer...

Reply:
There are a couple of related problems:

*   The `CPU_VARIANT`s are not very standardised, so the variant that we want depends on the `CPU` as well; and

*   The `CPU` / `CPU_VARIANT` are both set *before* we load anything from the `TARGET` Makefile, which means it's tool late to set a *default* `CPU_VARIANT` in the `TARGET` Makefile

Hence this work around, of setting it in the top level Makefile.  I think I've figured out a (slightly ugly, but hidden away in the `TARGET` Makefile) work around that is sufficient for the TinyFPGA BX / lm32 / minimal case.  I'm testing that now, and will push it if it builds okay.

Ewen

Reply:
I ran into this problem (via https://github.com/timvideos/litex-buildenv/issues/125) and fixed it see https://github.com/enjoy-digital/litex/pull/175 and https://github.com/enjoy-digital/litex/issues/159 . 

We just need to pull in the new litex.

Reply:
I *think* you're addressing the second test (on the `CPU` being the `DEFAULT_CPU`), rather than the `PLATFORM` test (since the `PLATFORM` test is gone now it's in the target Makefile.  But even with the names standardised (which is a great step forward!), it's possible that, eg, only lm32.minimal fits, but vexrisc.SOMETHING (else larger) will fit.  So I think the default variants required are still somewhat CPU specific.  (Ideally we'd probably set `DEFAULT_FULL_CPU`, and then split that apart.)

Ewen